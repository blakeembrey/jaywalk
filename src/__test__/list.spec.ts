import { validationFunction } from '../'

describe('List', () => {
  it('should validate lists', () => {
    const validate = validationFunction({
      '@type': 'List',
      items: {
        '@type': 'String'
      }
    })

    const input = ['a', 'b', 'c']

    return validate(input).then(function (output) {
      expect(output).toEqual(input)
    })
  })

  it('should reject non-lists', () => {
    const validate = validationFunction({
      '@type': 'List',
      items: {
        '@type': 'String'
      }
    })

    expect.assertions(1)

    return validate([1, 2, 3]).catch(function (err) {
      expect(err.errors).toMatchSnapshot()
    })
  })
})
