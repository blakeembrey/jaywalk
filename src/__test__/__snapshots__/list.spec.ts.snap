// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should reject non-lists 1`] = `
Array [
  Object {
    "dataPath": Array [
      0,
    ],
    "message": "not a string",
    "schemaPath": Array [
      "items",
      "@type",
    ],
  },
  Object {
    "dataPath": Array [
      1,
    ],
    "message": "not a string",
    "schemaPath": Array [
      "items",
      "@type",
    ],
  },
  Object {
    "dataPath": Array [
      2,
    ],
    "message": "not a string",
    "schemaPath": Array [
      "items",
      "@type",
    ],
  },
]
`;
